src/pip/_internal/req/constructors.py

```diff
@@ -137,16 +137,17 @@ def parse_editable(editable_req):
     vc_type = url.split('+', 1)[0].lower()
 
     if not vcs.get_backend(vc_type):
-        error_message = 'For --editable=%s only ' % editable_req + \
-            ', '.join([backend.name + '+URL' for backend in vcs.backends]) + \
-            ' is currently supported'
+        error_message = 'For --editable={} only '.format(
+            editable_req + ', '.join(
+                [backend.name + '+URL' for backend in vcs.backends]) +
+            ' is currently supported')
```

Comment:
I am having trouble parsing the parentheses, but they look wrong 🤔

Reply:
what is expected here..?

Reply:
@uranusjr Will this do..?
```
msg = ', '.join([backend.name + '+URL' for backend in vcs.backends])
error_message = 'For --editable={} only '.format(
            editable_req + msg + ' is currently supported')
raise InstallationError(error_message)
```

Reply:
The original relied on the precedence of `%` over `+`. I think the correct version would be:

```python
error_message = "For --editable={} only ".format(editable_req) + \
    ", ".join([backend.name + '+URL' for backend in vcs.backends]) + \
    "is currently supported"
```

And honestly, I still think that's ugly. At a minimum, I'd add a comma after the `{}` for readability. And I'd personally take that list of supported backends out:

```python
supported_backends = ", ".join([backend.name + '+URL' for backend in vcs.backends])
error_message = "For --editable={}, only {} is currently supported".format(
    editable_req, supported_backends
)
```

All I'd say for my revised version is that it's not horrible, and it's within the scope of a simple "change `%` to `format()`" PR. The whole message should be reviewed and rewritten IMO (the "is" should be "are" for multiple items, for example).

Reply:
@deepak1725 Nearly - see my version, you got the precedence of `%` and `+` backwards in your version, but otherwise your approach works for me.